Fix for gcc6 which does not have std::isfinite() but suggests
to use std::finite() instead.

--- JuceLibraryCode/modules/juce_core/maths/juce_MathsFunctions.h	2016-08-30 06:24:27.000000000 +0200
+++ JuceLibraryCode/modules/juce_core/maths/juce_MathsFunctions.h	2016-12-12 14:33:29.170005576 +0100
@@ -381,7 +381,7 @@
    #if JUCE_WINDOWS && !JUCE_MINGW
     return _finite (value) != 0;
    #else
-    return std::isfinite (value);
+    return finite (value);
    #endif
 }
 
@@ -391,7 +391,7 @@
    #if JUCE_WINDOWS && !JUCE_MINGW
     return _finite (value) != 0;
    #else
-    return std::isfinite (value);
+    return finite (value);
    #endif
 }
 
